server/internal/infrastructure/mongo/model.go

```diff
@@ -84,6 +85,24 @@ func (r *Model) FindByKey(ctx context.Context, projectID id.ProjectID, key strin
 	})
 }
 
+func (r *Model) FindByIDOrKey(ctx context.Context, modelID *id.ModelID, key *key.Key, pids id.ProjectIDList) (*model.Model, error) {
+	filters := make([]bson.M, 0, 2)
+	if modelID != nil {
+		filters = append(filters, bson.M{"id": modelID.String()})
+	}
+	if key != nil {
+		filters = append(filters, bson.M{"key": key.String()})
+	}
+	if len(filters) == 0 {
+		return nil, rerror.ErrNotFound
+	}
+
+	return r.findOne(ctx, bson.M{
+		"project": bson.M{"$in": pids.Strings()},
+		"$or":     filters,
```

Comment:
Maybe we need to use `$and` to insure the premissions

Reply:
Actually it works. See e2e test